src/Workspaces/Core/Portable/CodeRefactorings/CodeRefactoringContext.cs

```diff
@@ -48,14 +62,16 @@ public struct CodeRefactoringContext
         /// Add supplied <paramref name="action"/> to the list of refactorings that will be offered to the user.
         /// </summary>
         /// <param name="action">The <see cref="CodeAction"/> that will be invoked to apply the refactoring.</param>
-        public void RegisterRefactoring(CodeAction action)
+        public void RegisterRefactoring(CodeAction action) => RegisterRefactoring(action, null);
+
+        internal void RegisterRefactoring(CodeAction action, TextSpan? applicableToSpan)
```

Comment:
This should eventually be made a public API, whenever you are ready. Also why is the text span a nullable value type parameter? Callers should always pass in a valid applicable span OR use the overload without the span. You can probably have a shared private method that has signature and is invoked from both of them

We should also write an analyzer at https://github.com/dotnet/roslyn-analyzers/tree/master/src/Microsoft.CodeAnalysis.Analyzers/Core/FixAnalyzers that flags use of the prior overload without span if the second overload is available. Can you file a feature request issue on roslyn-analyzers repo for this?

Reply:
The reason for using nullable was that this channel needs to be able to carry CodeActions without the `TextSpan` (i.e. when the original overload is used) as well. There's nothing preventing us from representing missing `TextSpan` as `default` instead of `null` but thought `null` was more clear.

Reply:
I think the point is, you should have these signatures:

```
RegisterRefactoring(CodeAction)
RegisterRefactoring(CodeAction, TextSpan)
```

INternally, these can sit on top of a private:

```RegisterRefactoring(CodeAction, TextSpan?)```

Reply:
Ah, I see. That's doable but the nullability would still leak via the constructor and the saved lambda and it generally doesn't end up very clean. 

Representing missing `applicableToSpan` as default is an overall better solution.

Reply:
> Ah, I see. That's doable but the nullability would still leak via the constructor

Why?

Reply:
You're right, wrong assumption. Fixed.